### ISSUE TITLE
Import from Sketch

### ISSUE THREAD

**Issue by davidfuchsa** [Pixofield/keyshape-plugins#2]
In future it would be really helpful to be able to import sketch files. 

Are you planning to add it?

**Comment by badcat** [comment]
Hmm - Doesn't Sketch export directly to SVG? 

I believe both Gravit Designer and Figma have free accounts that would allow you to convert a Sketch file to SVG. YMMV :)
https://designer.io/
http://figma.com/

**Comment by davidfuchsa** [comment]
@badcat Yeah, they of course do. But all of the exports also messes up your layers, make a lot of defs etc. It would be nice to transfer your file 1:1 as you are used to work with it in your favorite tool. 

anyway, it works good with SVG. You only need to restructure it, rename all groups and layers to make sense of it etc, othjerwise you will be lost in a huuuuuuge mess creating more comlicated animations.

**Comment by davidfuchsa** [comment]
Its more about svg fromat itself and the difference between sketch file, you see?. But anyway, thanks for the cleaner, Ill try it for cleaning svg icons :)
![difference](https://user-images.githubusercontent.com/25434515/43846611-0f343fdc-9b2f-11e8-90cc-ea8b3d437517.jpg)

**Comment by badcat** [comment]
Ahh..  yes.. I see now.  Do you have a sample sketch file (and the default exported SVG) that has custom naming, grouping etc that I could play with?  Or point me to this somewhere on the web?  Interested to see the format.

**Comment by Pixofield** [comment]
Sketch has lots of features which are not supported by SVG or Keyshape, such as multiple fills and strokes, inner and outer stroke positions, symbol overrides and non-destructive boolean operations. It would take some time to create a mapping from them to SVG so that Keyshape can show them. It also means that there's no way to get a 1:1 mapping between Sketch and Keyshape layers.

Sketch files are zip files, which contain json data. However, there's no public documentation about all the fields in the json format. Also, the Sketch importer would need to be updated every time Bohemian Coding decides to release a new version of Sketch with new or changed json fields.

Sketch 50 seems to export the layer names as SVG ids and Keyshape is able show them in the Object Tree. If you are running an older version of Sketch, then upgrading it should help getting those layer names visible in Keyshape.

Creating a Sketch importer is possible, but it would take lots of time, so it is not under consideration at the moment.
